### DISCUSSION TITLE
aws-cdk/s3assets - deploying .zip files inside a folder structure of cdk bucket

### DISCUSSION THREAD

**Discussion by RajeshAWSGIT** [aws/aws-cdk#25081] [General]
I need to asset the .zip files from my local directory to a folder structure inside the bucket created by cdk.

code:-
asset= assets.Asset(self, 'SampleSingleFileAsset', path=os.path.join(dirname, 'cdk_s3poc'))

current - output :- 

New cdk bucket created - cdk-hnb659fds-assets-890725536181-eu-west-1 and .zip file deployed.

cdk-hnb659fds-assets-890725536181-eu-west-1 ----> .zip file deployed

Expected - output :- 

cdk-hnb659fds-assets-890725536181-eu-west-1 ----> macro folder -----> .zip file deployment

do we have any way to deploy the .zip files in the folder as mentioned above

**Comment by peterwoodworth** [comment]
Our `Asset` Api makes it easy to upload an asset to the bootstrap bucket - but it's not designed for this kind of control. We don't expose the ability to set the key 

You will need to use `BucketDeployment` to be able to upload a folder as-is
